presto-docs/src/main/sphinx/develop/verifier-mechanics.rst

```diff
@@ -0,0 +1,84 @@
+==================
+Verifier Mechanics
+==================
+
+The Presto Verifier can be used to test two versions Presto against each other.
+
+Procedures and Functionalities
+------------------------------
+
+The verifier verifies a set of given queries in the following procedures.
+
+Importing Source Queries
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Source queries can be read through MySQL.
+* If the required schema is not available, one can also plugin a custom Provider.
+
+Credentials and Filtering
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Catalog/schema/username/password overrides are applied if specified.
+* Whitelist and blacklist are applied if specified.
+* Syntactically invalid queries are filtered.
+* Custom filters can be plugged in if needed.
+
+Query rewriting
+~~~~~~~~~~~~~~~
+* Each query is rewritten before execution for setup purposes and also to ensure that verification does not interfere with production data.
```

Comment:
The purpose of setup is to ensure that we do not interfere with production data. So simplify this as `Each query is rewritten before execution to ensure that verification does not interfere with production data`